### ISSUE TITLE
MQTTMessageLogger can generate a null pointer exception when fixedHeader is null

### ISSUE THREAD

**Issue by windbender** [moquette-io/moquette#311]
this line:
```
                MqttMessageType messageType = msg.fixedHeader().messageType();
```
will generate an NPE on occasion.  This shows up over in NettyMQTTHandler, but the exact location is hard to track down.  Suggest wrapping  MQTTMessageLogger with something like 
```
             if(msg == null) {
                LOG.info("Trying logMQTT with null message for client id {}", clientID);
            } else if(msg.fixedHeader() == null) {
                LOG.info("Trying logMQTT with message with null fixed Header for client id {}", clientID);
            } else {
                .... old stuff here.
           }
```

**Comment by windbender** [comment]
~~I believe this may happen as a consequence of a internalPublish()  sourced message.~~

actually this appears to happen when the message delivered from Netty is an ERROR message. In my particular case it was because the message length was too long. (see another issue for that ).

So NettyMqttHandler.channelRead  should be prepared to handle failures on encoded within msg.encodeResult()'s response.

**Comment by andsel** [comment]
Fixed by #417
